File: X/rust/XRPL_UKL1/soal4.rs
/*
Soal          : 4
Gambar        : B
Matrix kolom  : 5
*/

fn main() {
    let a = 6;
    let b = 3;
    let kolom = 5;
    let mut u = a;
    let mut s = b;

    println!("Deret Aritmatikanya adalah");

    for i in (0..kolom).rev() {
        for _ in 0..=i {
            print!("{}\t", u);
            u += b;
            s += u;
        }
        println!();
    }

    for i in (1..kolom).rev() {
        for _ in 0..(kolom - i + 1) {
            print!("{}\t", u);
            u += b;
            s += u;
        }
        println!();
    }

    let total=s - u;
    println!("Jumlah Deret Aritmatikanya adalah {}", total);
}

File: X/javascript/XRPL_UKL1/soal4.js
/*
Soal          : 4
Gambar        : b
Matrix kolom  : 5
*/

a = 6;
b = 3;
kolom = 5;
u = a;
s = b;
console.log("Deret aritmatikanya adalah");

for (i = kolom; i > 0; i--) {
    for (j = 0; j < i; j++) {
        process.stdout.write(u + "\t");
        u += b;
        s += u;
    }
    console.log();
}

for (i = kolom - 1; i > 0; i--) {
    for (j = kolom + 1; j > 0 + i; j--) {
        process.stdout.write(u + "\t");
        u += b;
        s += u;
    }
    console.log();
}

total = s - u;
console.log("Jumlah deret aritmatikanya adalah " + total);
